### PR TITLE
Episodes 4 and 5 for issue #515

### PR DESCRIPTION
**Pull request by carschno**
Fixes for episodes 4 and 5 in issue #515

### CODE REVIEW

File: episodes/4-advanced-layer-types.Rmd [hunk 1]
```diff
@@ -690,18 +690,20 @@ The intuition behind dropout is that it enforces redundancies in the network by
...
 
-Let us add a dropout layer after each pooling layertowards the end of the network, that randomly drops 80% of the nodes.
+Let us add a dropout layer after each pooling layer towards the end of the network that randomly drops 80% of the nodes.
 
 ```python
 def create_nn_with_dropout():
     inputs = keras.Input(shape=train_images.shape[1:])
     x = keras.layers.Conv2D(50, (3, 3), activation='relu')(inputs)
     x = keras.layers.MaxPooling2D((2, 2))(x)
+    x = keras.layers.Dropout(0.8)(x) # This is new!
```
Comment (svenvanderburg) [inline episodes/4-advanced-layer-types.Rmd:702]: Strange.... how did this happen? Anyway, looks good!

File: episodes/5-transfer-learning.Rmd [hunk 2]
```diff
@@ -17,14 +17,16 @@ Instead of training a model from scratch, with transfer learning you make use of
 
 An example: Let's say that you want to train a model to classify images of different dog breeds. You could make use of a pre-trained network that learned how to classify images of dogs and cats. The pre-trained network will not know anything about different dog breeds, but it will have captured some general knowledge of, on a high-level, what dogs look like, and on a low-level all the different features (eyes, ears, paws, fur) that make up an image of a dog. Further training this model on your dog breed dataset is a much easier task than training from scratch, because the model can use the general knowledge captured in the pre-trained network.
 
+![](episodes/fig/05-transfer_learning.png)
+<!-- # TODO: perhaps the Mermaid markdown can be embedded directly in the Rmd file? -->
```
Comment (svenvanderburg) [inline episodes/5-transfer-learning.Rmd:21]: I like how we solve this in https://github.com/carpentries-incubator/python-intermediate-development/blob/da6fc57abee81115403dacce5ffc544ecf595a36/episodes/10-section1-intro.md?plain=1#L36-L54

We add a comment with instructions and everything you need to adapt the figure with mermaid.

Perhaps we can also do it in this way?

File: episodes/5-transfer-learning.Rmd [hunk 3]
```diff
@@ -37,6 +39,11 @@ val_images = np.load(DATA_FOLDER / 'test_images.npy')
 train_labels = np.load(DATA_FOLDER / 'train_labels.npy')
 val_labels = np.load(DATA_FOLDER / 'test_labels.npy')
 ```
+## 2. Identify inputs and outputs
```
Comment (svenvanderburg) [inline episodes/5-transfer-learning.Rmd:42]: Clearer!

File: episodes/5-transfer-learning.Rmd [hunk 4]
```diff
@@ -53,8 +60,9 @@ inputs = keras.Input(train_images.shape[1:])
 ```
 
 Our images are 64 x 64 pixels, whereas the pre-trained model that we will use was 
-trained on images of 160 x 160 pixels. To deal with this, we add an upscale layer
-that resizes the images to 160 x 160 pixels during training and prediction.
+trained on images of 160 x 160 pixels.
+To adapt our data accordingly, we add an upscale layer that that resizes the images to 160 x 160 pixels during training and prediction.
```
Comment (svenvanderburg) [inline episodes/5-transfer-learning.Rmd:64]:
```suggestion
To adapt our data accordingly, we add an upscale layer that resizes the images to 160 x 160 pixels during training and prediction.
```

File: episodes/5-transfer-learning.Rmd (hunk 2 above)
Comment (svenvanderburg) [reply episodes/5-transfer-learning.Rmd:21]: Really nice to add this picture by the way!

**Review by svenvanderburg — APPROVED** [review]
Great additions! Just a few comments, feel free to merge afterwards.
